scripts/generator/graphql_generator/csharp/SDK/DesktopCheckout.cs.erb

```diff
@@ -0,0 +1,44 @@
+#if UNITY_IOS
+namespace <%= namespace %>.SDK.iOS {
+    using System;
+    using System.Runtime.InteropServices;
+    using UnityEngine;
+    using <%= namespace %>.SDK;
+
+    class DesktopCheckout: WebCheckout {
+
+        private ShopifyClient _client;
+        private Cart _cart;
+
+        private static DesktopMessageReceiver _desktopMessageReceiver;
+
+        protected override ShopifyClient Client {
+            get {
+                return _client;
+            }
+        }
+
+        protected override Cart Cart {
+            get {
+                return _cart;
+            }
+        }
+
+        public DesktopCheckout(Cart cart, ShopifyClient client) {
+            _cart = cart;
+            _client = client;
+        }
+
+        public override void Checkout(string checkoutURL, CheckoutSuccessCallback success, CheckoutCancelCallback cancelled, CheckoutFailureCallback failure) {
+            #if !SHOPIFY_MONO_UNIT_TEST
+            if (_desktopMessageReceiver == null) {
+                _desktopMessageReceiver = GlobalGameObject.AddComponent<DesktopMessageReceiver>();
+            }
+            _desktopMessageReceiver.Init(_client, _cart.CurrentCheckout, success, cancelled, failure);
```

Comment:
I feel a bit uneasy about having `DesktopCheckout` not use `SetupNativeMessageReceiver`. Creating a `_desktopMessageReceiver` should ideally be set up there to make things consistent and clear.